File: 1. 단계별로_풀어보기/8. 수학 1단계/2_11005.js
// 입력
const fs = require('fs')
const inputArr = fs.readFileSync(0, 'utf-8').toString().trim().split('\n')

// 입력 파싱
let [N, B] = inputArr[0].split(' ').map(Number)

// 결과 문자열
let result = ''

while (N > 0) {
  // 나머지를 구한다.
  const M = N % B

  // M이 숫자인지, 문자인지 판단하고 result 에 덧붙인다.
  if (M < 10) {
    result = M.toString() + result
  } //
  else {
    result = String.fromCharCode(M - 10 + 'A'.charCodeAt(0)) + result
  }

  N = Math.floor(N / B)
}

process.stdout.write(`${result}`)

File: 1. 단계별로_풀어보기/1~9/8. 수학 1단계/2_11005.py
import sys, math

input = sys.stdin.readline
output = sys.stdout.write

N, B = map(int, input().split(' '))

result = ''

while N > 0:
  M = N % B

  if M < 10:
    result = str(M) + result
  else:
    result = chr(M - 10 + ord('A')) + result

  N = math.floor(N / B)

output(f"{result}")